### PR TITLE
Finish enabling unnecessary assignment analyzer

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Now is enabled as a warning again and will fail the command line build.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12534)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlCodeDomSerializer.cs [hunk 1]
```diff
@@ -334,45 +334,58 @@ private static Type[] ToTargetTypes(object context, Type[] runtimeTypes)
...
+        CodeStatementCollection statements,
+        object control,
+        string methodName,
+        CodeExpressionCollection? parameters,
+        Type[] paramTypes,
+        StatementOrdering ordering)
     {
-        string? name = manager.GetName(control);
+        // Not doing anything with the name, keeping the access for compat.
+        _ = manager.GetName(control);
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlCodeDomSerializer.cs:347]: Are these autofixes or is it required to have the `_ =`?

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DocumentDesigner.cs [hunk 2]
```diff
@@ -877,69 +876,57 @@ private void OnUndoing(object source, EventArgs e)
...
-                {
-                    if (cd.Control is not Form form || !form.TopLevel)
-                    {
-                        addControl = false;
-                    }
-                }
-            }
+        IDesigner designer = host.GetDesigner(component);
+        bool addControl = designer is ToolStripDesigner
+            || (designer is ControlDesigner cd && cd.Control is Form form && form.TopLevel);
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DocumentDesigner.cs:893]: I think this needs to be
```suggestion
        bool addControl = designer is ToolStripDesigner
            || designer is not ControlDesigner cd
            || (cd.Control is Form form && form.TopLevel);
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DocumentDesigner.cs [hunk 3]
```diff
@@ -877,69 +876,57 @@ private void OnUndoing(object source, EventArgs e)
...
-            if (addControl &&
-                TypeDescriptor.GetAttributes(component).Contains(DesignTimeVisibleAttribute.Yes))
+        if (!addControl || !TypeDescriptor.GetAttributes(component).Contains(DesignTimeVisibleAttribute.Yes))
+        {
+            return;
+        }
+
+        if (_componentTray is null)
+        {
+            if (TryGetService(out ISplitWindowService sws))
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DocumentDesigner.cs:902]: Consider combining ifs

**Review by Tanya-Solyanik — APPROVED** [review]

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlCodeDomSerializer.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlCodeDomSerializer.cs:347]: Required as it isn't actually used anywhere. `GetName` is observable so I'm not removing the call in this change.

**Review by lonitra — APPROVED** [review]

**Pull request merged**